nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -0,0 +1,1211 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+#pylint:disable=too-many-lines
+import warnings
+from collections import Counter
+from collections import OrderedDict
+from collections import deque
+from copy import deepcopy
+from enum import Enum
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
+
+import networkx as nx
+
+from nncf.common.graph import MODEL_INPUT_OP_NAME
+from nncf.common.graph import MODEL_OUTPUT_OP_NAME
+from nncf.common.graph import NNCFNodeName
+from nncf.common.graph import OperatorMetatype
+from nncf.common.graph.transformations.commands import TargetPoint
+from nncf.common.hardware.config import HWConfig
+from nncf.common.insertion_point_graph import InsertionPointGraph
+from nncf.common.quantization.quantizer_propagation.grouping import QuantizersWaitingForMergeManager
+from nncf.common.quantization.quantizer_propagation.structs import PropagatingQuantizer
+from nncf.common.quantization.quantizer_propagation.structs import QuantizationTrait
+from nncf.common.quantization.quantizer_propagation.graph import QuantizerPropagationStateGraph
+from nncf.common.quantization.quantizer_propagation.structs import QuantizerPropagationStateGraphNodeType
+from nncf.common.quantization.quantizer_setup import MultiConfigQuantizerSetup
+from nncf.common.quantization.quantizer_setup import QuantizationPointId
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizerSetup
+from nncf.common.quantization.structs import QuantizableWeightedLayerNode
+from nncf.common.quantization.structs import QuantizationConstraints
+from nncf.common.quantization.structs import QuantizationMode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.quantization.structs import QuantizerGroup
+from nncf.common.quantization.structs import UnifiedScaleType
+from nncf.common.utils.helpers import matches_any
+from nncf.common.utils.logger import logger as nncf_logger
+from nncf.common.utils.backend import __nncf_backend__
+
+if __nncf_backend__ == 'Torch':
+    # pylint:disable=wildcard-import
+    # pylint:disable=unused-wildcard-import
```

Comment:
Please, fix pylint errors.

Reply:
All of the entries below would have to be imported then, with the only purpose of appearing in the dict. There would be 30 more import lines. Is this really what you want.

Reply:
I don't suggest adding 30 more import lines. I guess you can import the module.

Reply:
Imported the module.